### PR TITLE
[#54, #55] Enhanced vocabularies management

### LINKED ISSUE #54: User // Add an existing vocabulary to a workspace
User should be able to pick from a list of existing vocabularies and add those to the workspace.

### LINKED ISSUE #55: User // Create a new vocabulary in a workspace
User should be able to create a brand new vocabulary within a workspace

### PR DESCRIPTION
**Pull request by karelklima**
Resolves #54  
Resolves #55

### CODE REVIEW

File: src/workspaces/components/AddVocabularyForm.tsx [hunk 1]
```diff
@@ -1,23 +1,72 @@
...
+  },
+  {
+    label: 'Legislativní',
+    prefix: 'https://slovník.gov.cz/legislativní/sbírka/',
+    regex: '^https://slovník.gov.cz/legislativní/sbírka/[0-9]+/[0-9]+$',
+  },
+  {
+    label: 'Generický',
+    prefix: 'https://slovník.gov.cz/generický/',
+    regex: '^https://slovník.gov.cz/generický/[a-z0-9]+$',
```
Comment (psiotwo) [inline src/workspaces/components/AddVocabularyForm.tsx:39]: Please change to  (my fault):

 ^https://slovník.gov.cz/generický/[ěščřžýáíéóúůďťňa-z0-9]+$

**Review by psiotwo — APPROVED** [review]
The code works as expected. Creating new vocabularies is not fully supported yet on server side, see https://github.com/opendata-mvcr/sgov/issues/61.

Might be more fluent to have a single "Add vocabulary" button, a single dialog, and let the user search the vocabulary first. If (s)he does not find it (e.g. "Create new" as a last item in the select box) allow the user to create a new one.

LGTM, approving (see one inline comment) - we should discuss the workflow.

File: src/workspaces/components/AddVocabularyForm.tsx (hunk 1 above)
Comment (karelklima) [reply src/workspaces/components/AddVocabularyForm.tsx]: I modified all the regular expressions to also allow "-" as a character in the last part of URI

**Comment by karelklima** [comment]
@psiotwo I combined the importing and creating of vocabularies feature into a single dialog as discussed.

The creating part supports custom label as well - it sends it as a "label" query string parameter which should be (hopefully) consistent with the rest of the API design, I can adjust that if necessary. But the bottom line is that once the server is fixed so that it accepts the custom label parameter, it should all work without any changes in the control panel.

**Review by psiotwo — APPROVED** [review]
lgtm

**Pull request merged**